firmware/gameport-adapter/Saitek.h

```diff
@@ -0,0 +1,71 @@
+// This file is part of Necroware's GamePort adapter firmware.
+// Copyright (C) 2021 Necroware
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include "AnalogJoystick.h"
+#include "Joystick.h"
+
+class Saitek : public Joystick {
+public:
+  const Description &getDescription() const override {
+    static const Description description{"Saitek", 3, 6, 1};

```

Comment:
Which Saitek Model exactly is this? There are a lot of Saitek devices and probably they are not all the same regarding amount of buttons, axes and hats. Also decoding could vary, I guess.

Reply:
The Saitek SP550 seems to have a (6 button) gameport version:
http://www.saitek.com/manuals/sp550four.pdf

As well as a (8 button?) USB version:
http://www.saitek.com/manuals/sp550usbeng.pdf

So don't get confused when searching for pictures online
The windows 95 VXD drivers (https://driver.ru/?H=SP550%20Gameport&By=Saitek) may contain interesting details, as the .inf file mentions a "ButtonMap" and "POVOMap", as well as:

`MFTR="Saitek Plc."
; we don't rename the stick key because the stick driver hasn't changed, and the key name is hard coded in the driver`